main.rb

```diff
@@ -0,0 +1,141 @@
+      yield(arr.at(i))
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    return enum_for unless block_given?
+
+    arr = to_a if is_a?(Range) || is_a?(Hash)
+    while i < arr.size
+      yield(arr.at(i), i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    selected = []
+    my_each { |item| selected << item if yield(item) }
+    selected
+  end
+
+  def my_all?(args = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+    elsif args.nil?
+      my_each { |item| return false if item.nil? || item == false }
+    elsif !args.nil? && (args.is_a? Class)
+      my_each { |item| return false if item.class != args }
+    elsif !args.nil? && instance_of?(Hash)
+      return false unless empty?
+    elsif !args.nil? && args.instance_of?(Regexp)
+      my_each { |item| return false unless args.match(item.to_s) }
+    else
+      my_each { |item| return false if item != args }
+    end
+    true
+  end
+
+  def my_any?(args = nil)
+    arr = to_a
+    unless arr.empty?
+      if block_given?
+        my_each { |item| return true if yield(item) }
+      elsif !args.nil? && (args.is_a? Class)
+        my_each { |item| return true if item.instance_of?(args) }
+      elsif !args.nil? && instance_of?(Hash)
+        return false unless empty?
+      elsif !args.nil? && args.instance_of?(Regexp)
+        my_each { |item| return true if args.match(item.to_s) }
+      elsif args.nil?
+        my_each { |item| return true if item == args }
+      else
+        my_each { |item| return true if item }
+      end
+    end
+    false
+  end
+
+  def my_none?(args = nil)
+    arr = to_a
+    unless arr.empty?
+      if block_given?
+        my_each { |item| return false if yield(item) }
+      elsif !args.nil? && (args.is_a? Class)
+        my_each { |item| return false if item.instance_of?(args) }
+      elsif !args.nil? && args.instance_of?(Regexp)
+        my_each { |item| return false if args.match(item.to_s) }
+      elsif !args.nil?
+        my_each { |item| return false if item == args }
+      else
+        my_each { |item| return false if item }
+      end
+    end
+    true
+  end
+
+  def my_count(args = nil)
+    counter = 0
+    if block_given?
+      my_each { |item| counter += 1 if yield(item) }
+      return counter
+    elsif !args.nil?
+      my_each { |item| counter += 1 if item == args }
+      return counter
+    end
+    size
+  end
+
+  def my_inject(num = 0, sym = nil)
+    result = num
+    unless block_given?
+      if num.instance_of?(Symbol) || num.instance_of?(String)
+        result = 0
+        my_each { |item| result = item.send(num, result) }
+      elsif !sym.nil? && sym.instance_of?(Symbol)
+        my_each { |item| result = item.send(sym, result) }
+      end
+      return result
+    end
+    return to_a if is_a?(Hash)
+
+    my_each { |item| result = yield(result, item) }
+    result
+  end
```

Comment:
- [ ] my_inject searches for the longest word in an array of strings.
```
  search = proc { |memo, word| memo.length > word.length ? memo : word }
  p %w[dog door rod blade].my_inject(&search)
```
The above code should return `blade`. Check with the original inject.

- [ ] my_inject when a block is given without an initial value combines all elements of enum by applying a binary operation,
```
  range = Range.new(5, 50) 

 actual = range.my_inject { |prod, n| prod * n }
 expected = range.inject { |prod, n| prod * n }
```
expected result: `1267253883404724085150525340252698701849068398706688000000000000`
gotten `0`